### PR TITLE
Roberto/reserved ip checker

### PR DESCRIPTION
**Pull request by roberto-graylog**
## Description
Added a helper class to help check whether an IP address is a Reserved IP Address (e.g. 192.168.x.x, 127.0.0.x).

## Motivation and Context
The need to check whether an IP address is a reserved address has arisen in several recent code changes.  This helper class in this location will lend itself for re-use in several places such as GeoIP processor and Greynoise adapters.

## How Has This Been Tested?
The code change includes a resource file with all the reserved IP subnets/blocks, which is used in the unit tests to confirm whether a set of address is/is not reserved.

## Screenshots (if appropriate):
N/A
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/test/java/org/graylog2/utilities/ReservedIpCheckerTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+package org.graylog2.utilities;
+
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ReservedIpCheckerTest {
+
+    @Test
+    public void testEsEmpty() {
```
Comment (kingzacko1) [inline graylog2-server/src/test/java/org/graylog2/utilities/ReservedIpCheckerTest.java:27]: Typo es -> is?

Comment (roberto-graylog) [reply graylog2-server/src/test/java/org/graylog2/utilities/ReservedIpCheckerTest.java:27]: Indeed.  Thanks for the catch.

Comment (kingzacko1) [reply graylog2-server/src/test/java/org/graylog2/utilities/ReservedIpCheckerTest.java]: No prob. All else looks good

**Comment by roberto-graylog** [comment]
I made a royal mess of this branch so, being such a small change, I just started this new branch [#11900 11900](https://github.com/Graylog2/graylog2-server/pull/11900)

**Pull request closed without merging**